### PR TITLE
moved hdf read/write

### PR DESCRIPTION
**Pull request by JoschD**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # TFS-Pandas Changelog
 
+## Version 3.2.0
+
+- Added: 
+    - HDF5 read/write. 
+
```
Comment (fsoubelet) [inline CHANGELOG.md:7]: Let's also say that the required Python version was bumped to `3.7`.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-18.04, ubuntu-20.04, macos-latest, windows-latest]
+        os: [ubuntu-18.04, ubuntu-20.04, ubuntu-22.04, macos-latest, windows-latest]
         # Make sure to escape 3.10 with quotes so it doesn't get interpreted as float 3.1 by GA's parser
         python-version: [3.7, 3.8, 3.9, "3.10"]
```
Comment (fsoubelet) [inline .github/workflows/tests.yml:19]: I guess we can add `3.11` here too

File: tfs/hdf.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+"""
+HDF5 Tools
```
Comment (fsoubelet) [inline tfs/hdf.py:2]: I would rather name this `HDF5 I/O` so the section in the documentation is extra clear.

File: tfs/hdf.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+----------
+
+Additional tools for reading and writing TfsDataFrames into hdf5 files.
```
Comment (fsoubelet) [inline tfs/hdf.py:5]: Double backtick around TfsDataFrames

File: tfs/hdf.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    """
+    _check_imports()
+    # Check for `key` kwarg (forbidden) ---
+    if "key" in kwargs:
+        raise AttributeError("The argument 'key' is not allowed here, "
+                             "as only one TfsDataFrame per file is supported.")
+
+    # Check for `mode` kwarg (allowed under circumstances but generally ignored) ---
+    user_mode = kwargs.pop('mode', None)
+    if user_mode is not None and user_mode != "w":
```
Comment (fsoubelet) [inline tfs/hdf.py:48]: I guess this should also check that the file already exists to be consistent with the docstring? Or should we modify the docstring to say "please don't provide mode, it has to be `w` anyway (which we hard-code just below)"?

File: tfs/hdf.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    return TfsDataFrame(df, headers=headers)
+
+
+def _check_imports():
+    not_imported = [name for name, package in (('tables', tables), ('h5py', h5py)) if package is None]
+    if len(not_imported):
+        names = ", ".join(f"`{name}`" for name in not_imported)
+        raise ImportError(f"Package(s) {names} could not be imported. "
+                          "Please make sure that this package is installed to use hdf-functionality, "
+                          "e.g. install `tfs-pandas` with the `hdf5` extra-dependencies: `tfs-pandas['hdf5']`")
```
Comment (fsoubelet) [inline tfs/hdf.py:90]: I'm not sure the " are needed for the extra specification, but that might be shell specific.

**Review by fsoubelet — COMMENTED** [review]
Minor comments here and there, I can make these changes if you wish. In terms of functionality and tests I am happy with this!

File: tfs/hdf.py (hunk 6 above)
Comment (JoschD) [reply tfs/hdf.py:90]: I think they are not. my bad

File: tfs/hdf.py (hunk 5 above)
Comment (JoschD) [reply tfs/hdf.py:48]: It checks that the file exist in the line below and raises an error if it does

File: .github/workflows/tests.yml (hunk 2 above)
Comment (fsoubelet) [reply .github/workflows/tests.yml]: Damn I don't know why I thought it was already out. My bad.

**Review by fsoubelet — APPROVED** [review]

**Pull request merged**